### PR TITLE
{bio}[GCCcore/9.3.0] SeqLib v1.2.0, fermi-lite v20190320, SSW v1.1

### PR DESCRIPTION
**Pull request by jfgrimm**
(created using `eb --new-pr`)

BWA will need to be rebuilt with the [updated easyblock](https://github.com/easybuilders/easybuild-easyblocks/pull/2417)

### CODE REVIEW

File: easybuild/easyconfigs/f/fermi-lite/fermi-lite-20190320-GCCcore-9.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    '46978098bb439d4772f263375eb67621489bb48853c182d7aab9b9da8a41494c',  # fermi-lite-20190320_build-shared-lib.patch
+]
+
+builddependencies = [('binutils', '2.34')]
+
+dependencies = [('zlib', '1.2.11')]
+
+files_to_copy = [
+    (['fml-asm'], 'bin'),
+    (['*.a', '*.so*'], 'lib'),
```
Comment (Micket) [inline easybuild/easyconfigs/f/fermi-lite/fermi-lite-20190320-GCCcore-9.3.0.eb:34]:
```suggestion
    (['*.a', '*.%s* % SHLIB_EXT'], 'lib'),
```

File: easybuild/easyconfigs/s/SSW/SSW-1.1-GCCcore-9.3.0.eb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+dependencies = [('zlib', '1.2.11')]
+
+start_dir = 'src'
+
+buildopts = 'default'
+
+files_to_copy = [
+    (['ssw_test'], 'bin'),
+    (['*.so*'], 'lib'),
```
Comment (Micket) [inline easybuild/easyconfigs/s/SSW/SSW-1.1-GCCcore-9.3.0.eb:39]:
```suggestion
    (['*.%s* % SHLIB_EXT'], 'lib'),
```

File: easybuild/easyconfigs/s/SSW/SSW-1.1_build-cpp-lib.patch [hunk 3]
```diff
@@ -0,0 +1,71 @@
+diff -Nru SSW.orig/src/Makefile SSW/src/Makefile
```
Comment (Micket) [inline easybuild/easyconfigs/s/SSW/SSW-1.1_build-cpp-lib.patch:1]: Could you add a short description of what the patch does, and preferably who wrote it (or a reference to a PR/issue)  (same for all the patches)

File: easybuild/easyconfigs/f/fermi-lite/fermi-lite-20190320-GCCcore-9.3.0.eb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    (['*.h'], 'include/fml'),
+    'test',
```
Comment (Micket) [inline easybuild/easyconfigs/f/fermi-lite/fermi-lite-20190320-GCCcore-9.3.0.eb:36]: is it not possible to run the tests?

File: easybuild/easyconfigs/f/fermi-lite/fermi-lite-20190320-GCCcore-9.3.0.eb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+]
+
+sanity_check_paths = {
+    'files': ['bin/fml-asm', 'lib/libfml.a', 'lib/libfml.%s' % SHLIB_EXT],
+    'dirs': [],
+}
+
```
Comment (Micket) [inline easybuild/easyconfigs/f/fermi-lite/fermi-lite-20190320-GCCcore-9.3.0.eb:43]: If possible, can you add a sanity_check_commands that does at least something like `fml-asm --version` or something?

File: easybuild/easyconfigs/s/SeqLib/SeqLib-1.2.0-GCC-9.3.0.eb [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    ('PCRE', '8.44'),
+]
+
+preconfigopts = "autoreconf -i -f && "
+
+sanity_check_paths = {
+    'files': ['bin/seqtools', 'lib/libseqlib.%s' % SHLIB_EXT],
+    'dirs': [],
+}
+
```
Comment (Micket) [inline easybuild/easyconfigs/s/SeqLib/SeqLib-1.2.0-GCC-9.3.0.eb:47]: a sanity_check_commands for `seqtools --version/--help` (if available)

**Review by Micket — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/f/fermi-lite/fermi-lite-20190320-GCCcore-9.3.0.eb (hunk 1 above)
Comment (jfgrimm) [reply easybuild/easyconfigs/f/fermi-lite/fermi-lite-20190320-GCCcore-9.3.0.eb:34]: These are actually not quite equivalent - `'*.so*'` will copy both `libfml.so` and `libfml.so.1`, whereas `'*.%s*' % SHLIB_EXT` only copies `libfml.so`.

File: easybuild/easyconfigs/f/fermi-lite/fermi-lite-20190320-GCCcore-9.3.0.eb (hunk 5 above)
Comment (jfgrimm) [reply easybuild/easyconfigs/f/fermi-lite/fermi-lite-20190320-GCCcore-9.3.0.eb:43]: Unfortunately, `fml-asm` doesn't look for `--version`, `--help` or similar, and providing no arguments prints the usage but exits with status code 1

File: easybuild/easyconfigs/f/fermi-lite/fermi-lite-20190320-GCCcore-9.3.0.eb (hunk 4 above)
Comment (jfgrimm) [reply easybuild/easyconfigs/f/fermi-lite/fermi-lite-20190320-GCCcore-9.3.0.eb:36]: the suggested way to run the test is `fml-asm test/MT-simu.fq.gz > MT.fq`, however there is nothing to verify the output. I could add something like `fml-asm %(installdir)s/test/MT-simu.fq.gz > /dev/null` to `sanity_check_commands`?

Comment (Micket) [reply easybuild/easyconfigs/f/fermi-lite/fermi-lite-20190320-GCCcore-9.3.0.eb:36]: I'd say it's not worth it in that case.

File: easybuild/easyconfigs/f/fermi-lite/fermi-lite-20190320-GCCcore-9.3.0.eb (hunk 1 above)
Comment (Micket) [reply easybuild/easyconfigs/f/fermi-lite/fermi-lite-20190320-GCCcore-9.3.0.eb:34]: But I included the * in at the end?

Comment (jfgrimm) [reply easybuild/easyconfigs/f/fermi-lite/fermi-lite-20190320-GCCcore-9.3.0.eb:34]: wow, not sure how I missed that. I even wrote it in my response and it still didn't register

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node031.pri.viking.alces.network - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz, Python 2.7.5
See https://gist.github.com/9799571c5196e166f96f6fbcafc3c53f for a full test report.

**Comment by Micket** [comment]
@boegelbot please test @ generoso

**Comment by jfgrimm** [comment]
It's failing on `#include "bwa/bwa.h"` because BWA hasn't been rebuilt against the updated easyblock

**Comment by Micket** [comment]
@boegelbot please test @ generoso
EB_ARGS="--rebuild BWA-0.7.17-GCC-9.3.0.eb fermi-lite-20190320-GCCcore-9.3.0.eb SSW-1.1-GCCcore-9.3.0.eb SeqLib-1.2.0-GCC-9.3.0.eb "

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
bear-pg0211u08b.bear.cluster - Linux RHEL 8.3, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/6353f7d43735e8d97614f98c9f57435d for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
bear-pg0211u03a.bear.cluster - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.8.5
See https://gist.github.com/8016711d88f68bdf25145f3baea064e6 for a full test report.

**Comment by branfosj** [comment]
Going in, thanks @jfgrimm!

**Pull request merged**